src/go/accessd/config/testdata/accessd_mconfig.json

```diff
@@ -0,0 +1,16 @@
+// comment outside of json
+{
```

Comment:
Lots to say here, which I learned last week in the config debug on master.

The below is intended to describe how the Magma AGW presently operates, per my understanding from discussions with @ardzoht and review of [magma dev documentation](https://docs.magmacore.org/docs/lte/dev_notes).

## mconfig files

As part of Magma install, a single [gateway.mconfig](https://sourcegraph.com/github.com/magma/magma/-/blob/lte/gateway/configs/gateway.mconfig?subtree=true) is placed at `/etc/magma/gateway.mconfig`. Note that it has a special shape and expects an initial JSON key of `configs_by_key` - then an array of per-daemon protobuf representations with a `"@type"` annotation.

If you were going to use mconfig for your configuration, you would modify this static configuration...

But - big but - as part of check-in with an `Orc8r` instance, `Orc8r` will push new configs to the AGW. This new `mconfig` is then stored ephemerally to `/var/opt/magma/configs/gateway.mconfig` and once present completely disregards any contents of your static `/etc/magma/gateway.mconfig`.

So if you are going to go the `mconfig` route, you have to wire this configuration change in through `NMS UI` and `Orc8r config push` all the way to the mconfg write to disk.. Otherwise the first Orc8r push will over-write your values.

There is an alternative...

## YAML Service Configs

So for local AGW service behavior configuration, the intended path today is to write a YAML service config to `/etc/magma/<service_name>.yml`.

I am unclear whether any pre-defined schema exists for these files, I think it does not. So they are totally at-will w.r.t. additions / expectation of content.  You would then read that configuration YAML and operate on it at runtime, for any fields it defines.